### PR TITLE
Allow subclassing of AttrsInstance and Protocol at once

### LINKED ISSUE #1133: AttrsInstance is only a Protocol in mypy
`__init__.pyi` defines AttrsInstance as follows:
```
class AttrsInstance(AttrsInstance_, Protocol):
    pass
```

So the following typechecks correctly:
```
import attr
from typing import Protocol

class Foo(attr.AttrsInstance, Protocol):
    def attribute(self) -> int:
        ...


def bar(foo: Foo) -> int:
    return foo.attribute()

@attr.define
class CFoo:
    a: int

    def attribute(self) -> int:
        return self.a

bar(CFoo(15))
```

But if you run it you get:

`TypeError: Protocols can only inherit from other protocols, got <class 'attr.AttrsInstance'>`

Because in `__init__.py` it's defined as:
```
class AttrsInstance:
    pass
```

instead of 
```
class AttrsInstance(Protocol):
    pass
```

### PR DESCRIPTION
**Pull request by hynek**
Fixes #1133

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
           python -Im coverage html --skip-covered --skip-empty
 
           # Report and write to summary.
-          python -Im coverage report | sed 's/^/    /' >> $GITHUB_STEP_SUMMARY
+          python -Im coverage report --format=markdown >> $GITHUB_STEP_SUMMARY
```
Comment (euresti) [inline .github/workflows/ci.yml:96]: This isn't related to the change right?

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,4 @@ tmp*
 attrs.docset
 attrs.tgz
 Justfile
+t.py
```
Comment (euresti) [inline .gitignore:22]: What's this for?

**Review by euresti — APPROVED** [review]
Looks good. Thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hynek) [reply .github/workflows/ci.yml:96]: yes, sorry that snook in

File: .gitignore (hunk 2 above)
Comment (hynek) [reply .gitignore:22]: The place I tested your types. ;) Hatch is a bit weird about what files it packages, so I've stopped using global gitignores / .git/info/exclude for projects that I release myself.

**Pull request merged**